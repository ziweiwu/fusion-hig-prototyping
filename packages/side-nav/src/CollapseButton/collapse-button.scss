@import "~@hig/styles/mixins/base.scss";
@import "~@hig/styles/mixins/colors.scss";

.hig__side-nav__module__collapse-button {
  @include vendor-prefix("transform", "rotate(0deg)");

  &:hover {
    cursor: pointer;
  }
}

.hig__side-nav__module__collapse-button--collapsed {
  @include vendor-prefix("transform", "rotate(-90deg)");
}

// Light theme
.hig--light-theme {
  &.hig__side-nav__module__collapse-button {
    svg * {
      stroke: color(hig-cool-gray-50);
    }
  }
}

// Dark blue theme
.hig--dark-blue-theme {
  &.hig__side-nav__module__collapse-button {
    svg * {
      stroke: color(hig-white);
    }
  }
}
