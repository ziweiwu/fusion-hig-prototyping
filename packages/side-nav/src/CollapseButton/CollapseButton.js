import React, { Component } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import Icon, { names as iconNames, sizes as iconSizes } from "@hig/icon";
import { ThemeContext } from "@hig/themes";

import "./collapse-button.scss";

/** @todo Consider replacing with an IconButton */
export default class CollapseButton extends Component {
  static propTypes = {
    /** Presents the icon in a minimized state: caret pointing right */
    minimized: PropTypes.bool,
    /** Called when element is clicked */
    onClick: PropTypes.func
  };

  static defaultProps = {
    onClick: () => {},
    minimized: false
  };

  render() {
    const { minimized, onClick } = this.props;
    const classes = themeClass =>
      cx(themeClass, "hig__side-nav__module__collapse-button", {
        "hig__side-nav__module__collapse-button--collapsed": minimized
      });

    return (
      <ThemeContext.Consumer>
        {({ themeClass }) => (
          /** @todo replace with a semantic HTML tag */
          <div
            className={classes(themeClass)}
            onClick={onClick}
            role="button"
            tabIndex={0}
          >
            <Icon name={iconNames.CARET} size={iconSizes.PX_24} />
          </div>
        )}
      </ThemeContext.Consumer>
    );
  }
}
