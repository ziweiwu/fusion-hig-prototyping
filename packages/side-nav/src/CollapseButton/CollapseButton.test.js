import React from "react";
import renderer from "react-test-renderer";

import CollapseButton from "./CollapseButton";

describe("side-nav/CollapseButton", () => {
  describe("snapshot tests", () => {
    const cases = [
      {
        description: "renders with no props",
        props: {}
      },
      {
        description: "renders minimized",
        props: {
          minimized: true
        }
      }
    ];

    cases.forEach(({ description, props: { children, ...otherProps } }) => {
      it(description, () => {
        const wrapper = (
          <CollapseButton {...otherProps}>{children}</CollapseButton>
        );
        const tree = renderer.create(wrapper).toJSON();

        expect(tree).toMatchSnapshot();
      });
    });
  });
});
