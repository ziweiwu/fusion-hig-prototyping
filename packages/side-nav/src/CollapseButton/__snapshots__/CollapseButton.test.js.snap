// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`side-nav/CollapseButton snapshot tests renders minimized 1`] = `
<div
  className="hig--light-theme hig__side-nav__module__collapse-button hig__side-nav__module__collapse-button--collapsed"
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="hig__icon hig__icon--24-size"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg width=\\"10\\" height=\\"6\\" viewBox=\\"0 0 10 6\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"caret-24\\"><polyline transform=\\"rotate(45 9.828 3)\\" points=\\"8 2 8 8 2 8\\" stroke=\\"#2E4258\\" stroke-width=\\"1.15\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"/></svg>",
      }
    }
  />
</div>
`;

exports[`side-nav/CollapseButton snapshot tests renders with no props 1`] = `
<div
  className="hig--light-theme hig__side-nav__module__collapse-button"
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="hig__icon hig__icon--24-size"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg width=\\"10\\" height=\\"6\\" viewBox=\\"0 0 10 6\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"caret-24\\"><polyline transform=\\"rotate(45 9.828 3)\\" points=\\"8 2 8 8 2 8\\" stroke=\\"#2E4258\\" stroke-width=\\"1.15\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"/></svg>",
      }
    }
  />
</div>
`;
